core/services/workflows/engine.go

```diff
@@ -669,10 +695,18 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 	// receiving loop may not pick up any messages we emit.
 	// Note: When full persistence support is added, any hanging steps
 	// like this one will get picked up again and will be reprocessed.
+	e.stepUpdatesChMapMu.RLock()
+	stepUpdateCh, ok := e.stepUpdatesChMap[stepState.ExecutionID]
+	if !ok {
+		e.stepUpdatesChMapMu.RUnlock()
```

Comment:
Can we encapsulate calls to the map in something which enforces the calls to the lock for us?

eg:
```
type channelsMap struct {
    m map[string]stepUpdateChannel
    mu sync.RWLock
}

func (c *channelsMap) get(executionID) stepUpdateChannel {
    // acquire lock, defer unlock
    // do your logic
}
```